src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -642,7 +648,8 @@ public static MsBuildToolset GetMsBuildFromMonoPaths(string userVersion)
                 switch (userVersion)
                 {
                     case "14.1": return new MsBuildToolset(version: "14.1", path: msBuildPathOnMono14);
-                    case "15":
+                    case var version when string.Equals(version, "latest", StringComparison.OrdinalIgnoreCase):
```

Comment:
Instead of doing it this way, I would rather you mirror the implementation in `GetMsBuildToolset` and either set `userVersion` to `string.Empty`, or just do `if (string.IsNullOrEmpty(userVersion) || string.Equals(userVersion, "latest", StringComparison.OrdinalIgnoreCase)` over here:
https://github.com/NuGet/NuGet.Client/blob/5564b076b3055a16b1765b3abd3f67025d484a28/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs#L639-L645

I believe this will be much clearer. Your other change seems good, though!

Reply:
Done, thanks for the suggestion!